### ISSUE TITLE
Crash renaming file from outside the project

### ISSUE THREAD

**Issue by duarteroso** [godotengine/godot#60379]
### Godot version

3.4.4.stable

### System information

Windows 10, macOS

### Issue description

The editor crashes when renaming files that are not directly under the project root.
For example, my shared code is linked to the project via symlinks and therefore does not live inside the godot `res://` path.
Renaming a script will crash the editor each time.

The crash only seems to happen when the project explorer is in `Split Mode` with the folder and files separated (see screenshot attached)

![screenshot](https://user-images.githubusercontent.com/11413364/164085871-63459245-010f-4ab3-a78a-b48ffd04342a.PNG)

### Steps to reproduce

Please find attached a repro project.
The zip contains:
1) A folder with external source files
1) A Godot project with a symlink to the external source folder

To reproduce the crash:
1. Open the project
1. Make sure the solution explorer is in Split Mode (see screenshot above).
1. Search for a file (ex: `Scrip`, see screenshot above)
1. Rename that file
1. Editor will crash

### Minimal reproduction project

[rename_bug.zip](https://github.com/godotengine/godot/files/8516175/rename_bug.zip)

**Comment by Calinou** [comment]
I can't reproduce this on `master` 1d2177938. Am I following the steps correctly?

https://user-images.githubusercontent.com/180032/164118368-3904028c-1962-4130-8bbc-654d6e84a33b.mp4

**Comment by duarteroso** [comment]
It also happens when renaming a `.gd` script file.

The zip file actually breaks the symlink folder (surprisingly!) and duplicates the folder it was pointing to (surprisingly!!)
Seems that archiving to `7z` then to `zip` does preserve the symlink 🥳  Let me know if this worked better or if you are on a mac I'll provide a proper archive
[rename_bug_symlink.zip](https://github.com/godotengine/godot/files/8524896/rename_bug_symlink.zip)

Basically it should look like this
![screen1](https://user-images.githubusercontent.com/11413364/164311792-dbff732f-f7d2-4596-bd40-1d5475b6faa2.PNG)

Cheers!
